server/services/invidious.ts

```diff
@@ -0,0 +1,416 @@
+        for (const tag of tags) {
+          const attrs = Object.fromEntries(
+            Array.from(tag.matchAll(/(\w+)=["']([^"']+)["']/g)).map(m => [m[1], m[2]])
+          ) as Record<string, string>;
+          const bw = Number(attrs["bandwidth"]);
+          if (!bw || Number.isNaN(bw)) continue;
+          const w = attrs["width"]  ? Number(attrs["width"])  : undefined;
+          const h = attrs["height"] ? Number(attrs["height"]) : undefined;
+          const codecs = attrs["codecs"];
+          reps.push({ bw, w: (w && !Number.isNaN(w)) ? w : undefined, h: (h && !Number.isNaN(h)) ? h : undefined, codecs });
+        }
+        reps.sort((a, b) => b.bw - a.bw);
+        const top = reps.slice(0, 5).map(r => ({
+          bandwidth_kbps: Math.round(r.bw / 1000),
+          resolution: (r.w && r.h) ? `${r.w}x${r.h}` : undefined,
+          codecs: r.codecs,
+        }));
+        log.info(
+          `DASH manifest summary url=${manifestUrl} total=${reps.length} variants=${JSON.stringify(top)}`
+        );
+        if (reps.length === 0) {
+          const ct = String((resp.headers as any)['content-type'] || '');
+          log.warn(
+            `DASH manifest parsed 0 variants (content-type=${ct}). First 300 chars: ${text.slice(0, 300)}`
+          );
+        }
+      } else {
+        // HLS master: #EXT-X-STREAM-INF lines
+        const lines = text.split(/\r?\n/);
+        const entries: Array<{ bw?: number; res?: string; codecs?: string }> = [];
+        for (let i = 0; i < lines.length; i++) {
+          const line = lines[i];
+          if (line.startsWith("#EXT-X-STREAM-INF:")) {
+            const bw = Number((line.match(/BANDWIDTH=(\d+)/) || [])[1]);
+            const res = (line.match(/RESOLUTION=(\d+x\d+)/) || [])[1];
+            const codecs = (line.match(/CODECS="([^"]+)"/) || [])[1];
+            entries.push({ bw: Number.isNaN(bw) ? undefined : bw, res, codecs });
+          }
+        }
+        entries.sort((a, b) => (b.bw || 0) - (a.bw || 0));
+        const top = entries.slice(0, 5).map(e => ({
+          bandwidth_kbps: e.bw ? Math.round(e.bw / 1000) : undefined,
+          resolution: e.res,
+          codecs: e.codecs,
+        }));
+        log.info(
+          `HLS master playlist summary url=${manifestUrl} total=${entries.length} variants=${JSON.stringify(top)}`
+        );
+        if (entries.length === 0) {
+          const ct = String((resp.headers as any)['content-type'] || '');
+          log.warn(
+            `HLS manifest parsed 0 variants (content-type=${ct}). First 300 chars: ${text.slice(0, 300)}`
+          );
+        }
+      }
+    } catch (e) {
+      log.warn(
+        `Failed to fetch/parse ${kind.toUpperCase()} manifest for logging url=${manifestUrl} err=${e instanceof Error ? e.message : e}`
+      );
+    }
+  }
+
+  /** Probe a manifest and return its top bitrate/resolution. 
+   * Needs to be there after finishing feature to give frontend all infos about quality
```

Comment:
AFAIK, the client should be able to figure out this info. I'm pretty sure Plyr gives us this info, similar to how it gives us info about captions. We shouldn't need to compute it on the server side.